src/main/java/org/elasticsearch/action/support/replication/TransportShardReplicationOperationAction.java

```diff
@@ -631,9 +631,22 @@ void performOnReplica(final ReplicationState state, final ShardRouting shard, fi
             }
 
             final ReplicaOperationRequest shardRequest = new ReplicaOperationRequest(shardIt.shardId(), state.replicaRequest());
+
+            // If the replicas use shadow replicas, there is no reason to
+            // perform the action on the replica, so skip it and
+            // immediately return
+            if (IndexMetaData.isIndexUsingShadowReplicas(indexMetaData.settings())) {
+                // this doesn't replicate mappings changes, so can fail if mappings are not predefined
+                // It was successful on the replica, although we never actually executed - in the future we will
```

Comment:
I would clarify this statement? cause mapping updates do get replicated, it just takes longer since it needs to head to the master and then published to the replicas, so there is a delay in mapping introduction.